### PR TITLE
[depends] use ccache if available

### PR DESCRIPTION
**Pull request by Rechi**
## Description
Detect ccache on depends configuration and if available use it for native and target depends. Also added an option to explicitly deactivate this feature (`--disable-ccache`).

I'm not sure if `CC`, `CXX` and `CPP` should have ccache in all this files (target/config.site, native/config.site.native, Makefile.include target/Toolchain.cmake, target/config-binaddons.site, target/Toolchain_binaddons.cmake).

## Motivation and Context
This will speed up the build of depends after you have done it once.

## How Has This Been Tested?
Build depends on OS X for OS X.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

maybe @wsnipex for review

### CODE REVIEW

File: tools/depends/Makefile.include.in [hunk 1]
```diff
@@ -44,10 +44,12 @@ CONFIG_GUESS=@prefix@/@tool_dir@/share/automake-1.15/config.guess
...
 
+USE_CCACHE=@use_ccache@
+
 LD=@LD@
-CC=@CC@
-CXX=@CXX@
-CPP=@CPP@
+CC=@CCACHE@ @CC@
+CXX=@CCACHE@ @CXX@
+CPP=@CCACHE@ @CPP@
```
Comment (wsnipex) [inline tools/depends/Makefile.include.in]: those should only be set like this is USE_CCACHE is true, also further down

**Review by wsnipex — COMMENTED** [review]
why not simply set CC and CXX to ccache in configure.ac and leave all the rest alone?

**Comment by Rechi** [comment]
>why not simply set CC and CXX to ccache in configure.ac and leave all the rest alone?

If you look at the different commits I have actually tried that in the commit `[depends] use ccache if available`.

The problem with this approach is, that cmake doesn't like compiler definitions with spaces in it. With cmake there must be set `RULE_LAUNCH_COMPILE` to ccache.
Another issue is that `target/Toolchain.cmake.in` is also used for compiling kodi and then you wouldn't be able to enable ccache for depends and disable ccache for kodi.

File: tools/depends/Makefile.include.in (hunk 1 above)
Comment (Rechi) [inline tools/depends/Makefile.include.in]: @wsnipex does this also apply to `native/config.site.native.in` and `target/config.site.in`?

**Comment by wsnipex** [comment]
ok, thanks for the explanation and fixup, looks good to me now.
jenkins build this with addons please

**Comment by Rechi** [comment]
jenkins build this please

**Pull request merged**